core/src/main/java/org/sonarsource/sonarlint/core/http/HttpClientManager.java

```diff
@@ -64,34 +60,14 @@ public class HttpClientManager {
 
   public HttpClientManager(SonarLintClient client, String userAgent, Path sonarlintUserHome) {
     this.client = client;
-    var sslFactoryBuilder = SSLFactory.builder()
+    var confirmingTrustManager = new ConfirmingTrustManager(sonarlintUserHome, client);
+    var sslFactory = SSLFactory.builder()
       .withDefaultTrustMaterial()
       .withSystemTrustMaterial()
-
-      .withTrustMaterial(new X509TrustManager() {
-
-        @Override
-        public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
-          // Not implemented
-        }
-
-        @Override
-        public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
-
-        }
-
-        @Override
-        public X509Certificate[] getAcceptedIssuers() {
-          return new X509Certificate[0];
-        }
-      });
-
-    var sonarlintTruststore = sonarlintUserHome.resolve("/ssl/truststore.jks");
-    if (Files.exists(sonarlintTruststore)) {
-      sslFactoryBuilder.withTrustMaterial(sonarlintTruststore, "changeit".toCharArray());
-    }
+      .withTrustMaterial(confirmingTrustManager)
+      .build();
     var asyncConnectionManager = PoolingAsyncClientConnectionManagerBuilder.create()
-      .setTlsStrategy(Apache5SslUtils.toTlsStrategy(sslFactoryBuilder.build()))
+      .setTlsStrategy(new DefaultClientTlsStrategy(sslFactory.getSslContext()))
```

Comment:
I'd suggest using
`new DefaultClientTlsStrategy(
                sslFactory.getSslContext(),
                sslFactory.getSslParameters().getProtocols(),
                sslFactory.getSslParameters().getCipherSuites(),
                SSLBufferMode.STATIC,
                sslFactory.getHostnameVerifier()
        )`
to fully use our sslFactory object parameters, and also to move from the default hostname verifier to the OkHttpClient one (copied and used by the library)

Reply:
Just noticed this comment. Yes the default hostname verifier of SSLFactory is a copy of the one from OkHttp